src/query/expression/src/types/binary.rs

```diff
@@ -331,19 +334,14 @@ impl BinaryColumnBuilder {
     }
 
     pub fn repeat(scalar: &[u8], n: usize) -> Self {
-        let len = scalar.len();
-        let mut data = Vec::with_capacity(len * n);
-        for _ in 0..n {
-            data.extend_from_slice(scalar);
-        }
-        let offsets = once(0)
-            .chain((0..n).map(|i| (len * (i + 1)) as u64))
-            .collect();
-        BinaryColumnBuilder {
-            data,
-            offsets,
+        let mut builder = BinaryColumnBuilder {
+            data: Vec::new(), // lazy allocate
```

Comment:
We already know the size of the `scalar` and the number of repetitions `n`, what is the benefit of lazy allocate ?

Reply:
`vec.extend` will always get the size from `iterator.size_hint` to allocate memory, there's no need to do it manually.
https://github.com/rust-lang/rust/blob/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/alloc/src/vec/mod.rs#L3008-L3016

Reply:
Why not just use `with_capacity`.